idioms/priv-extend.md

```diff
@@ -43,10 +49,27 @@ fn main(s: a::S) {
 }
 ```
 
+`#[non_exhaustive]` only works across crate boundaries. Within a crate, the private field method may be used:
+
+For `struct`s, an alternative approach exists: By adding a private field to a struct, the struct cannot be instantiated or matched against.
+
+```rust
+pub struct S {
+   pub a: i32,
+   // Because `b` is private, you cannot match on `S` without using `..` and `S` cannot be directly instantiated
+   b: ()
```

Comment:
```suggestion
    pub a: i32,
    // Because `b` is private, you cannot match on `S` without using `..` and `S` cannot be directly instantiated
    b: ()
```